### PR TITLE
Nahida Chiki completed assignment1

### PR DESCRIPTION
**Pull request by nchiki**

### CODE REVIEW

File: nchiki/assignment-1/Main.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    public static boolean isAnagram(String fst, String snd) {
+        int[] letters = new int[26];
+        int offset;
+        int lengthAlphabet;
+        offset = 9;
+        lengthAlphabet = 26;
+        for (int i = 0; i < fst.length(); i++) {
+            char letter = fst.charAt(i);
+            int numericVal = Character.getNumericValue(letter)-offset;
+            letters[numericVal]++;
```
Comment (TDiva) [inline nchiki/assignment-1/Main.java]: That's a nice approach, but what happens if there's a non-latin letter in the input string? 
Also, the idea of using array of integers is basicly the same as using a map, so maybe HashMap will be more convenient here?

File: nchiki/assignment-1/OptionalCaseSensitive.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+public class OptionalCaseSensitive {
+
+    //takes case of each letter into consideration
+    public static boolean isCaseSensitiveAnagram(String fst, String snd) {
```
Comment (TDiva) [inline nchiki/assignment-1/OptionalCaseSensitive.java]: I think the original version is case sensitive, let's do optional case insensitive. Also, consider using HashMap instead of array with symbols, because of the same reason.

File: nchiki/assignment-1/Tests.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import org.junit.Test;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+
+public class Tests {
```
Comment (TDiva) [inline nchiki/assignment-1/Tests.java]: It would be nice to have more test cases covering more complicated situations: empty input, big strings, additional characters to make sure code performs well in any conditions.

File: nchiki/assignment-1/Main.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+public class Main {
+
+
+    public static boolean isAnagram(String fst, String snd) {
+        HashMap<Character, Integer> letters = new HashMap<>();
+        initMap(letters);
+        for (int i = 0; i < fst.length(); i++) {
+            char letter = fst.charAt(i);
+            System.out.println(letter);
+      System.out.println(letters.get(letter));
```
Comment (TDiva) [inline nchiki/assignment-1/Main.java]: Please, format this file:
- Remove unnecessary empty lines
- Fix indentation.

File: nchiki/assignment-1/Main.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        for (int i = 0; i < snd.length(); i++) {
+            char letter = snd.charAt(i);
+            int val = letters.get(letter)-1;
+            letters.put(letter, val);
+        }
+        return checkHashMap(letters);
+    }
+
+    private static void initMap(HashMap<Character, Integer> map) {
+        int lengthSymbols = 94;
```
Comment (TDiva) [inline nchiki/assignment-1/Main.java]: Generally it's better to avoid "magic numbers": 
https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad

But here it's better to drop this "initialization" alltogether - you are reserving memory which might not be needed, why not do that when you actually need to store smth there?
